tests/test_point_range.py

```diff
@@ -0,0 +1,89 @@
+"""
+Tests for Point.
+
+To run this unit test install UnitTesting package and choose
+"UnitTesting: Test current project" command.
+"""
+
+from unittest import TestCase
+
+from LSP.main import Point
+from LSP.main import Range
+
+
+class test_point(TestCase):
+
+    def test_valid_point(self):
+        lsp_point = {'line': 10, 'character': 4}
+        point = Point.from_lsp(lsp_point)
+        self.assertEqual(point.row, 10)
+        self.assertEqual(point.col, 4)
+
+    def test_none_point(self):
+        # expect TypeError if range object is None
+        with self.assertRaises(TypeError):
+            Point.from_lsp(None)
```

Comment:
The tests against TypeError and KeyError confirm nothing about what LSP does, only built-in python behaviour. Please remove them.

Reply:
This is not completely true. As you call `.get()` methods for most/all dict objects of a language-server-respond even crucial elements such as the `line` and `character` part of a point, the resulting class would contain None in that place with unpredictable results. You may trust a language-server to always return valid objects, but I don't.

The method `from_lsp` does not always need to raise an exception. It would not if a None check was added to it. With other words it could return None instead of raising the exception. This check ensures the exception to be raised.